packages/dashboard/app/withdraw/utils/handlers.ts

```diff
@@ -0,0 +1,147 @@
+import { LIT_CHAINS } from '@lit-protocol/constants';
+
+let isLitNodeClientInitialized = false;
+let litNodeClient: LitNodeClient;
+
+async function initLitNodeClient() {
+  if (!isLitNodeClientInitialized) {
+    litNodeClient = new LitNodeClient({
+      litNetwork: SELECTED_LIT_NETWORK
+    });
+    await litNodeClient.connect();
+    isLitNodeClientInitialized = true;
+  }
+  return litNodeClient;
+}
+
+type ShowStatusFn = (message: string, type: StatusType) => void;
+
+// Standard ERC20 token ABI - just the functions we need
+const ERC20_ABI = [
+  "function symbol() view returns (string)",
+  "function decimals() view returns (uint8)",
+];
+
+export const handleSubmit = async (
+  isCustomToken: boolean,
+  tokenAddress: string,
+  withdrawAmount: string,
+  withdrawAddress: string,
+  agentPKP: IRelayPKP,
+  sessionSigs: SessionSigs,
+  chainId: string,
+  setLoading: (loading: boolean) => void,
+  showStatus: ShowStatusFn,
+) => {
+  if (!withdrawAmount || !withdrawAddress) {
+    showStatus('Please fill all fields', 'warning');
+    return { success: false };
+  }
+
+  if (parseFloat(withdrawAmount) === 0) {
+    showStatus('Withdrawal amount cannot be zero', 'warning');
+    return { success: false };
+  }
+
+  if (!ethers.utils.isAddress(withdrawAddress)) {
+    showStatus('Invalid withdrawal address', 'error');
+    return { success: false };
+  }
+
+  try {
+    setLoading(true);
+    showStatus('Preparing withdrawal...', 'info');
+    
+    const chain = LIT_CHAINS[chainId];
+    const rpcUrl = chain.rpcUrls?.[0];
+    
+    litNodeClient = await initLitNodeClient();
+
+    const pkpWallet = new PKPEthersWallet({
+      pkpPubKey: agentPKP.publicKey,
+      litNodeClient: litNodeClient,
+      controllerSessionSigs: sessionSigs,
+      rpc: rpcUrl
+    });
+
+    await pkpWallet.init();
+    
+    const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    
+    // Default token setup
+    let token = {
+      address: '',
+      symbol: '',
+      decimals: 18,
+    };
+
+    let transactionResult;
+    if (isCustomToken && ethers.utils.isAddress(tokenAddress)) {
+      showStatus('Fetching token details...', 'info');
+      
+      try {
+        const tokenContract = new ethers.Contract(tokenAddress, ERC20_ABI, provider);
+        const [symbol, decimals] = await Promise.all([
+          tokenContract.symbol(),
+          tokenContract.decimals()
+        ]);
+        
+        token = {
+          address: tokenAddress,
+          symbol,
+          decimals
+        };
+
+        const amount = ethers.utils.parseUnits(withdrawAmount, token.decimals);
+
+        transactionResult = await sendTokenTransaction({
+          pkpWallet,
+          tokenDetails: token,
+          amount,
+          recipientAddress: withdrawAddress,
+          provider
+        });
+        showStatus(`Detected token: ${symbol}`, 'info');
+      } catch (error) {
+        showStatus('Could not fetch token details.', 'error');
```

Comment:
```suggestion
        showStatus('Could not fetch token details. Check token address and try again.', 'error');
```
Good thing is that we get this error when trying to use a non ERC20 address